wlauto/workloads/iozone/__init__.py

```diff
@@ -0,0 +1,136 @@
+                                                             'iozone'))
+        Iozone.device_binary = self.device.install(Iozone.host_binary)
+
+    def setup(self, context):
+        self.results = os.path.join(self.device.working_directory,
+                                    iozone_results_txt)
+        self.command = self._build_command()
+
+        if self.threads and self.auto_mode:
+            raise ConfigError("You cannot set the number of threads and enable"
+                              " auto mode at the same time.")
+
+    def _build_command(self):
+        iozone_command = '{}'.format(self.device_binary)
+
+        if self.auto_mode:
+            self.auto_option = ' -a'
+            iozone_command += self.auto_option
+
+        if self.tests:
+            self.test_string = ''.join([' -i {}'.format(t) for t in self.tests])
+            iozone_command += self.test_string
+
+        if self.record_length > 0:
+            self.record_option = ' -r {}'.format(self.record_length)
+            iozone_command += self.record_option
+
+        if self.threads > 0:
+            self.threads_option = ' -t {}'.format(self.threads)
+            iozone_command += self.threads_option
+
+        if self.file_size > 0:
+            self.file_size_option = ' -s {}'.format(self.file_size)
+            iozone_command += self.file_size_option
+
+        if self.other_params:
+            other_params_string = ' ' + self.other_params
+            iozone_command += other_params_string
+
+        self.log_string = ' > {}'.format(self.results)
+        iozone_command += self.log_string
+
+        return iozone_command
+
+    def run(self, context):
+        self.device.execute(self.command, timeout=self.timeout)
+
+    def update_result(self, context):
+        self.device.pull_file(self.results, context.output_directory)
+        outfile = os.path.join(context.output_directory, iozone_results_txt)
+
+        with open(outfile, 'r') as iozone_file:
+            for line in iozone_file:
+                time_res_match = time_res_regex.search(line)
+                if time_res_match:
+                    context.result.add_metric("time_resolution",
+                                              float(time_res_match.group(1)),
+                                              time_res_match.group(2))
+
+                cache_size_match = cache_size_regex.search(line)
+                if cache_size_match:
+                    context.result.add_metric("processor_cache_size",
+                                              float(cache_size_match.group(1)),
+                                              cache_size_match.group(2))
+
+                cache_line_match = cache_line_regex.search(line)
+                if cache_line_match:
+                    context.result.add_metric("processor_cache_line_size",
+                                              float(cache_line_match.group(1)),
+                                              cache_line_match.group(2))
```

Comment:
What you said made sense, and I would have gone along with your suggestion. Except I'm running into the problem where if users execute the iozone workload with the -t option enabled (to run a specific number of threads), then the results would be different:


    Iozone: Performance Test of File I/O
            Version $Revision: 3.430 $
        Compiled for 32 bit mode.
        Build: linux-arm

    Contributors:William Norcott, Don Capps, Isom Crawford, Kirby Collins
                 Al Slater, Scott Rhine, Mike Wisner, Ken Goss
                 Steve Landherr, Brad Smith, Mark Kelly, Dr. Alain CYR,
                 Randy Dunlap, Mark Montague, Dan Million, Gavin Brebner,
                 Jean-Marc Zucconi, Jeff Blomberg, Benny Halevy, Dave Boone,
                 Erik Habbinga, Kris Strecker, Walter Wong, Joshua Root,
                 Fabrice Bacchella, Zhenghua Xue, Qin Li, Darren Sawyer,
                 Vangel Bojaxhi, Ben England, Vikentsi Lapa.

    Run began: Fri Aug 14 04:07:55 2015

    Command line used: /usr/local/bin/iozone -t 2
    Output is in kBytes/sec
    Time Resolution = 0.000001 seconds.
    Processor cache size set to 1024 kBytes.
    Processor cache line size set to 32 bytes.
    File stride size set to 17 * record size.
    Throughput test with 2 processes
    Each process writes a 512 kByte file in 4 kByte records

    Children see throughput for  2 initial writers  =  213579.16 kB/sec
    Parent sees throughput for  2 initial writers   =   16403.29 kB/sec
    Min throughput per process          =   95499.95 kB/sec
    Max throughput per process          =  118079.20 kB/sec
    Avg throughput per process          =  106789.58 kB/sec
    Min xfer                    =     412.00 kB

    Children see throughput for  2 rewriters    =  398145.08 kB/sec
    Parent sees throughput for  2 rewriters     =   20088.98 kB/sec
    Min throughput per process          =  172895.72 kB/sec
    Max throughput per process          =  225249.36 kB/sec
    Avg throughput per process          =  199072.54 kB/sec

Thoughts?